### PR TITLE
Metadata integration tests

### PR DESCRIPTION
**Pull request by JanKaul**
This PR adds integration tests for reading the table metadata from files. Some of the tests are designed to fail. With the current design of the serialization/deserialization the error doesn't specify which field is missing. So I couldn't do a precise check for certain tests.

### CODE REVIEW

**Comment by JanKaul** [comment]
@liurenjie1024, @Xuanwo , @Fokko it would be great if you could take a look.

File: crates/iceberg/src/spec/table_metadata.rs [hunk 1]
```diff
@@ -346,21 +349,29 @@ pub(super) mod _serde {
...
-                        .map(|schema| Ok((schema.schema_id, Arc::new(schema.try_into()?))))
-                        .collect::<Result<Vec<_>, Error>>()?,
-                ),
-                current_schema_id: value.current_schema_id,
+                current_schema_id: if schemas.keys().contains(&value.current_schema_id) {
+                    Ok(value.current_schema_id)
+                } else {
+                    Err(self::Error::new(
+                        ErrorKind::DataInvalid,
+                        "No schema exists with the current schema id.",
```
Comment (liurenjie1024) [inline crates/iceberg/src/spec/table_metadata.rs:371]:
```suggestion
                        format!("No schema exists with the current schema id: {}", *value.current_schema_id),
```

**Review by liurenjie1024 — APPROVED** [review]
LGTM, left a small suggestion. Thanks, @JanKaul !

**Review by Fokko — APPROVED** [review]
This is great @JanKaul, I see a lot of good stuff in this PR, thanks for adding these tests 👍🏻

**Pull request merged**